### PR TITLE
add voting::is_proposal_id_assigned

### PR DESCRIPTION
**Pull request by hamdiallam**
### Description

voting.move aborts on invalid proposal ids. In order for our governance module to present a cleaner error, a view function on assigned proposal ids would be useful

### Test Plan

n/a as this is simply a view function addition.

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/voting.move [hunk 1]
```diff
@@ -488,11 +488,19 @@ module aptos_framework::voting {
...
     /// Return the next unassigned proposal id
-    public fun next_proposal_id<ProposalType: store>(voting_forum_address: address,): u64 acquires VotingForum {
+    public fun next_proposal_id<ProposalType: store>(voting_forum_address: address): u64 acquires VotingForum {
         let voting_forum = borrow_global<VotingForum<ProposalType>>(voting_forum_address);
         voting_forum.next_proposal_id
     }
 
+    #[view]
+    /// Return true if the proposal id has been assigned
+    public fun is_proposal_id_assigned<ProposalType: store>(voting_forum_address: address, proposal_id: u64): bool acquires VotingForum {
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/voting.move:498]: nit: Rename this as proposal_exists

**Review by movekevin — DISMISSED** [review]
Why did boogie.bpl change so much?

File: aptos-move/framework/aptos-framework/sources/voting.move (hunk 1 above)
Comment (hamdiallam) [reply aptos-move/framework/aptos-framework/sources/voting.move:498]: will do

**Comment by hamdiallam** [comment]
> Why did boogie.bpl change so much?

Hm, i'm not sure but the additions look normal? I simply ran `cargo build -p aptos-cached-packages` after making the changes

**Comment by movekevin** [comment]
> > Why did boogie.bpl change so much?
> 
> Hm, i'm not sure but the additions look normal? I simply ran `cargo build -p aptos-cached-packages` after making the changes

cc @rahxephon89 Do you know why boogie.bpl would change so much? This PR somehow generates 193,163 additions into that file

**Comment by rahxephon89** [comment]
I don't think boogie.bpl should be added to the repo in the first place. @hamdiallam @movekevin

File: aptos-move/framework/aptos-framework/boogie.bpl
Comment (rahxephon89) [inline aptos-move/framework/aptos-framework/boogie.bpl:1]: I don't think it should be included in the PR.

Comment (hamdiallam) [reply aptos-move/framework/aptos-framework/boogie.bpl:1]: removed -- not sure where this came from

**Comment by hamdiallam** [comment]
bump :) @movekevin @rahxephon89

**Review by davidiw — APPROVED** [review]

**Review by movekevin — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/voting.spec.move [hunk 2]
```diff
@@ -175,6 +175,12 @@ spec aptos_framework::voting {
         aborts_if !exists<VotingForum<ProposalType>>(voting_forum_address);
     }
 
+    spec proposal_exists<ProposalType: store>(voting_forum_address: address, proposal_id: u64): bool {
+        use aptos_framework::chain_status;
+        requires chain_status::is_operating(); // Ensures existence of Timestamp
+        include AbortsIfNotContainProposalID<ProposalType>;
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/voting.spec.move:181]: this is wrong, your function cannot abort under these circumstances.

**Review by davidiw — CHANGES_REQUESTED** [review]

**Comment by davidiw** [comment]
closing out, feel free to re-open when fixed / complete.

**Pull request closed without merging**